sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -154,7 +196,8 @@ string IOptionsFormatter.Format()
                 {
                     { nameof(MaxBatchSize), MaxBatchSize },
                     { nameof(BatchCheckpointFrequency), BatchCheckpointFrequency },
-                    { nameof(ConnectionOptions), ConstructConnectionOptions() },
+                    { nameof(TransportType),  TransportType.ToString()},
+                    { nameof(WebProxy),  WebProxy?.GetProxy(new Uri("https://servicebus.windows.net")).ToString() ?? string.Empty },
```

Comment:
The issue is that we don't have the FQ namespace in the options. Also we wouldn't have this in the custom JSON converter. With the concrete `WebProxy` type it shouldn't matter - you just need to pass some destination URI and it will give back the proxy that was specified. If a customer is using some custom IWebProxy instance, then in theory it could spit out a different URI, but this is used for diagnostics only.

Reply:
/cc @pakrym for his thoughts as well

Reply:
I'm thinking that not formatting this value might be not that bad.

Reply:
Or we can only roundtrip it when the `WebProxy` is used.

Reply:
Well this is an EH/SB option - it is used outside of functions as well.

Reply:
Alright, will update to type sniff and round-trip only if the proxy is instance of WebProxy. Also, will try to rework the tests to avoid needing custom converter.